_posts/2019-11-01-deploying-debezium-with-kafkaconnector-resource.md

```diff
@@ -0,0 +1,697 @@
+    include.schema.changes: "true" 
+EOF
+```
+
+In `metadata.labels`, `strimzi.io/cluster` names the `KafkaConnect` cluster which this connector will be created in.
+
+The `spec.class` names the Debezium MySQL connector and `spec.tasksMax` can be 1 because that's all this connector ever uses.
+
+The `spec.config` object contains the rest of the connector configuration.
+The [Debezium documentation](https://debezium.io/documentation/reference/0.10/connectors/mysql.html#connector-properties) explains the available properties, but it's worth calling out some specifically:
+* I'm using `database.hostname: 192.168.99.1` as IP address for connecting to MySQL because I'm using `minikube` with the virtualbox VM driver
+  If you're using a different VM drvier with `minikube` you might need a different IP address.
+* The `database.port: "3306"` works because of the `-p 3306:3306` argument we used when we started up the MySQL server.
+* The `database.whitelist: "inventory"` basically tells Debezium to only watch the `inventory` database.
+* The `database.history.kafka.topic: "schema-changes.inventory"` configured Debezium to use the `schema-changes.inventory` topic to store the database schema history.
+
+A while after you've created this connector you can have a look at its `status`, using `kubectl get kctr inventory-connector -o yaml`:
+
+```
+...
+status:
+  conditions:
+  - lastTransitionTime: "2019-11-18T15:01:53.135Z"
+    status: "True"
+    type: Ready
+  observedGeneration: 1
+```
+
+This tells us that the connector is running within the `KafkaConnect` cluster we created in the last step.
+
+
+To summarise, we've now the complete picture with the connector talking to MySQL:
+
+<pre style="line-height: 1.2 !important;"><code>
+┌────────────────────────────────┐         ┌────────────────────────────────┐
+│ minikube, namespace: kafka     │         │ docker                         │
+│                                │         │                                │
+│ Kafka                          │     ┏━━━┿━▶ MySQL                        │
+│  name: my-cluster              │     ┃   │                                │
+│                                │     ┃   └────────────────────────────────┘
+│ KafkaConnect                   │     ┃
+│  name: my-connect              │     ┃
+│                                │     ┃
+│ KafkaConnector                 │     ┃
+│  name: inventory-connector ◀━━━┿━━━━━┛
+│                                │
+└────────────────────────────────┘
+</code></pre>
+
+> OK, I admit it: I was lying about the ASCII art being awesome.
+
+
+# Showtime!
+
+If you list the topics, for example using `kubectl exec my-cluster-kafka-0 -c kafka -i -t -- bin/kafka-topics.sh --zookeeper '' --list` you should see:
+
+```
+__consumer_offsets
+connect-cluster-configs
+connect-cluster-offsets
+connect-cluster-status
+dbserver1
+dbserver1.inventory.addresses
+dbserver1.inventory.customers
+dbserver1.inventory.geom
```

Comment:
That's not in the (4) tables you were showing before?

Reply:
I didn't add that table, but it's in Debezium's demo MySQL dockers images. Do you know what it's there for?

Reply:
Yes, it is to show off geospatial column types being captured. I was only wondering why it didn't show up in the four tables you listed somewhere before.

Reply:
Because I copy-and-pasted that bit from the Debezium tutorial. Now fixed.